routers/web/admin/users.go

```diff
@@ -336,7 +336,10 @@ func EditUserPost(ctx *context.Context) {
 
 	if len(form.UserName) != 0 && u.Name != form.UserName {
 		if err := user_setting.HandleUsernameChange(ctx, u, form.UserName); err != nil {
-			ctx.Redirect(setting.AppSubURL + "/admin/users")
+			if ctx.Written() {
```

Comment:
Why not redirect? The flash should be also displayed even after redirecting.

Reply:
`ctx.Written()` the response has been written, how to redirect?

Reply:
If redirect to user list, we can not continually edit the user, and need to find the user again and access the edit page again.